wipp-backend-data/src/main/java/gov/nist/itl/ssd/wipp/backend/data/csvCollection/CsvCollectionDataHandler.java

```diff
@@ -31,31 +33,34 @@
 public class CsvCollectionDataHandler  extends BaseDataHandler implements DataHandler{
 
 
-	@Autowired
-	CoreConfig config;
+    @Autowired
+    CoreConfig config;
 
-	@Autowired
-	private CsvCollectionRepository csvCollectionRepository;
+    @Autowired
+    private CsvCollectionRepository csvCollectionRepository;
 
-	@Override
-	public void importData(Job job, String outputName) throws JobExecutionException {
-		CsvCollection csvCollection = new CsvCollection(job, outputName);
-		csvCollectionRepository.save(csvCollection);
+    @Autowired
+    private CsvHandler csvHandler;
 
+    @Override
+    public void importData(Job job, String outputName) throws JobExecutionException, IOException {
+        CsvCollection csvCollection = new CsvCollection(job, outputName);
+        csvCollectionRepository.save(csvCollection);
 
-		File csvCollectionFolder = new File(config.getCsvCollectionsFolder(), csvCollection.getId());
-		csvCollectionFolder.mkdirs();
 
-		File tempOutputDir = getJobOutputTempFolder(job.getId(), outputName);
-		boolean success = tempOutputDir.renameTo(csvCollectionFolder);
-		if (!success) {
-			csvCollectionRepository.delete(csvCollection);
-			throw new JobExecutionException("Cannot move csv collection to final destination.");
-		}
+        File csvCollectionFolder = new File(config.getCsvCollectionsFolder(), csvCollection.getId());
+        csvCollectionFolder.mkdirs();
+
+        File tempOutputDir = getJobOutputTempFolder(job.getId(), outputName);
+        importFolder(csvCollectionFolder, csvCollection.getId());
+        boolean success = tempOutputDir.renameTo(csvCollectionFolder);
+        if (!success) {
+            csvCollectionRepository.delete(csvCollection);
+            throw new JobExecutionException("Cannot move csv collection to final destination.");
+        }
+        setOutputId(job, outputName, csvCollection.getId());
+    }
```

Comment:
There are duplicated operations here: `importFolder` is sufficient, the `renameTo` step is redundant, and `importFolder` should use `tempOutputDir` as input, not `csvCollectionFolder`. `csvCollectionFolder` is not needed here.
Something like:
```java
       try {
            File jobOutputTempFolder = getJobOutputTempFolder(job.getId(), outputName);
            csvHandler.importFolder(csvCollection.getId(), jobOutputTempFolder);
            setOutputId(job, outputName, csvCollection.getId());
        } catch (IOException ex) {
            csvCollectionRepository.delete(csvCollection);
            throw JobExecutionException("Cannot move CSV collection to final destination.");
        }
```